core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala

```diff
@@ -100,6 +101,7 @@ class JdbcRDD[T: ClassTag](
     }
 
     override def close() {
+      if (closed) return
       try {
         if (null != rs) {
           rs.close()
```

Comment:
From my understanding, we could call `rs = null` after it is closed, also for `stmt` and `conn` in the same way to make this function is reenterable.